packages/sdk/src/sdk.ts

```diff
@@ -365,31 +377,80 @@ export class NxtpSdk {
     }
 
     const inbox = generateMessagingInbox();
-    const receivedResponsePromise = new Promise<AuctionResponse>((res, rej) => {
+
+    let AuctionBids: AuctionResponse[] = [];
+
+    const receivedResponsePromise = await new Promise<AuctionResponse>((res, rej) => {
       setTimeout(() => rej(), 10_000);
-      this.messaging.subscribeToAuctionResponse(inbox, (data, err) => {
+      this.messaging.subscribeToAuctionResponse(inbox, async (data, err) => {
         if (err || !data) {
           this.logger.error({ method, methodId, err, data }, "Error in auction response");
-          return;
         }
-
         // dry run, return first response
-        if (!data.bidSignature) {
-          res(data);
+        else if (!data.bidSignature) {
+          AuctionBids.push(data);
           return;
+        } else {
+          AuctionBids.push(data);
+          if (AuctionBids.length >= 5) {
+            return;
+          }
         }
+      });
 
+      AuctionBids.sort((a, b) => {
+        return parseFloat(b.bid.amountReceived) - parseFloat(a.bid.amountReceived);
```

Comment:
We can't parseFloat here, we are dealing with BigNumbers that could be larger than JS numbers.